### ISSUE TITLE
Advanced API with Dingo, CORS and JWT

### ISSUE THREAD

**Issue by dudheit** [SaifurRahmanMohsin/RESTful-Plugin#2]
Hi,

firstly, thanks for your awesome work here! I gave your extension a quick look and it seems quite promising. As I haven't read through the code base I don't know how this actually works, but I was wondering whether, with this approach, it would be possible to integrate a sophisticated API (Dingo API based, Transformers, CORS and JWT protected) into OctoberCMS as well. 
I am actually aiming to integrate a Laravel Dingo API to OctoberCMS in this way. Can you give some advice whether this might work?

Thanks!

**Comment by SaifurRahmanMohsin** [comment]
Yep. In fact I may include transformers as part of this plugin in the near future (I'm currently busy writing a series of paid plugins and a mobile project using those plugins so it would take a while). I have not tried Dingo but I have already worked with jwt and it works fine. In fact i would be making a separate plugin (I think called Auth) to add the security layer to RESTful APIs.

**Comment by dudheit** [comment]
Great! I'll will fork and play around with it a bit then and see if I manage to integrate Dingo.

**Comment by dudheit** [comment]
I managed to integrate Dingo but I can't get JWTAuth running. When I use the middleware ``[ 'middleware' => '\Tymon\JWTAuth\Middleware\GetUserFromToken' ]`` I get a ``Class auth does not exist`` exception since October does not include the AuthServiceProvider. How did you manage to work around that?

**Comment by SaifurRahmanMohsin** [comment]
That’s weird. I was able to make it work without any issues when I wrote a private plugin a few months ago. I used the same `tymon/jwt-auth` package.

And what do you mean October doesn’t include AuthServiceProvider? I do remember using an Auth class. Besides when I wrote that plugin it worked very well. I will be writing an Auth plugin in the near future as well to support Auth via RESTful endpoints.

**Comment by dudheit** [comment]
Mhm, have a look at my routes.php with DingoAPI and JWT:

    \App::register('\Dingo\Api\Provider\LaravelServiceProvider');
    \App::register('Tymon\JWTAuth\Providers\JWTAuthServiceProvider'); 
 
    $api->version('v1', function ($api) {
          $api->group( [ 'middleware' => '\Tymon\JWTAuth\Middleware\GetUserFromToken' ], function ($api) {
               $api->resource('foo', 'Bar');
     ....

The middleware seems to raise the error ``Auth``  not found. Any ideas what is the problem?

**Comment by SaifurRahmanMohsin** [comment]
My apologies. I’m a little pre-occupied with work so I’m unable to explain it to you. Kindly check the code I posted [here](3#issuecomment-261109401). It may help.

**Comment by dudheit** [comment]
Right no worries. I will see whether I can find a solution. Thanks.
